test/e2e/e2e_suite_test.go

```diff
@@ -198,8 +205,58 @@ func TestMain(m *testing.M) {
 
 			return ctx, nil
 		},
+
+		func(ctx context.Context, cfg *envconf.Config) (context.Context, error) {
+			if containerTool == "podman" {
+				log.Println("deleting image archive")
+				if err := removeImageArchive(); err != nil {
+					log.Printf("failed to delete image archive: %s", err)
+				}
+			}
+			return ctx, nil
+		},
 	)
 
 	// Use Environment.Run to launch the test
 	os.Exit(testEnv.Run(m))
 }
+
+func useImageArchive(kindCluster *kind.Cluster, ctx context.Context) (context.Context, error) {
+	var err error
+	log.Println("Creating docker image archive...")
+	err = createImageArchive()
+	if err != nil {
+		log.Printf("Failed to create archive %s:", err)
+		return ctx, err
+	}
+	log.Println("Loading docker image archive...")
+	ctx, err = loadImageArchive(*kindCluster, ctx)
+	if err != nil {
+		return ctx, err
+	}
+	return ctx, nil
+}
+
+func createImageArchive() error {
+	// Save docker image archive
+	cmd := exec.Command("make", "docker-save", fmt.Sprintf("ARCHIVE_NAME=%s", archiveName))
```

Comment:
I tried running this locally with Podman, but I'm getting the following cleanup error:

```
2025/04/02 09:31:33 Destroying cluster...
2025/04/02 09:31:40 deleting image archive
2025/04/02 09:31:40 failed to delete image archive: remove etcdOperator.tar: no such file or directory
FAIL    go.etcd.io/etcd-operator/test/e2e       181.442s
FAIL
make: *** [Makefile:75: test-e2e] Error 1
```

Should we write the archive under a [`T.TempDir`](https://pkg.go.dev/testing#T.TempDir) directory? SO we don't need to do a cleanup later?

Reply:
I wasn't aware of T.TempDir, I will have a look into this

Reply:
I had a look at using T.TempDir but I wasn't sure how to use it within the scope of TestMain as that imports `*testing.M`, instead I used `os.MkdirTemp` and it is performing the cleanup function well.